### ISSUE TITLE
Update to websockets in ghcjs-base?

### ISSUE THREAD

**Issue by ariep** [Saulzar/ghcjs-websockets-reflex#1]
Hi!

This package depends on ghcjs-websockets, which does not compile under newest ghcjs: https://github.com/mstksg/ghcjs-websockets/issues/9. Among other changes, ghcjs-base now includes a Websocket module: https://github.com/ghcjs/ghcjs-base/blob/4b8ad4ebc95b77f1e89181bdd57101535182e810/JavaScript/Web/WebSocket.hs.

Do you think it is feasible to drop the dependency on ghcjs-websockets, and reimplement on top of the ghcjs-base version? It seems that the callback-type interface is suited to building a reflex-layer on top of it.

**Comment by oliver-batchelor** [comment]
Yeah, I think that might be a reasonable idea. Only trouble is that reflex-dom doesn't support the latest ghcjs-dom either!

Though there's the 'hamishmack' branch which does - and yes, I agree the more basic callback interface seems better for fitting in with reflex's Event lazy subscription.
